### PR TITLE
Refactored EventsNearMe to use async/await

### PR DESCRIPTION
**Pull request by dan8239**

### CODE REVIEW

File: lambda/custom/controllers/IntentController.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+const LocationHandler=require('../handlers/location/LocationHandler');
+const EventsHandler=require('../handlers/events/EventsHandler');
+
+
+exports.getEventsWithUserLocation = async function getEventsWithUserLocation(apiEndpoint, token, deviceID){
```
Comment (rknruben56) [inline lambda/custom/controllers/IntentController.js]: Should have a comment

File: lambda/custom/handlers/events/EventsHandler.js [hunk 2]
```diff
@@ -34,4 +35,21 @@ let searchEventbrite = (latitude, longitude, callback) => {
 	});
 };
 
+exports.asyncGetEventsNearUserLocation = async function asyncGetEventsNearUserLocation(latitude, longitude){
```
Comment (rknruben56) [inline lambda/custom/handlers/events/EventsHandler.js]: should have the "async" prefix removed and have a comment. searchEventsNearMe() function on line 12 can now be removed too

File: lambda/custom/handlers/location/LocationHandler.js [hunk 3]
```diff
@@ -40,4 +41,31 @@ exports.getLocation = (apiEndpoint, token, deviceId, callback) => {
         let locationString = locationProperties.filter((prop) => {return prop !== null}).join(",");
         geocoder.getLatLong(locationString, callback);
     });
-};
\ No newline at end of file
+};
+
+exports.asyncGetLocation = async function asyncGetLocation(apiEndpoint, token, deviceId){
```
Comment (rknruben56) [inline lambda/custom/handlers/location/LocationHandler.js]: should have a comment

File: lambda/custom/index.js [hunk 4]
```diff
@@ -46,11 +47,14 @@ const handlers = {
             this.emit(':tell', "implement nearest bus location intent");
         }       
     },
-	'EventLocationIntent': function() {		
+	'EventLocationIntent': async function() {		
         let parameters = ParameterHelper.getLocationParameters(this.event.context.System);
+        let alexaResponse = await IntentController.getEventsWithUserLocation(parameters.apiEndpoint, parameters.token, parameters.deviceID);
+        this.emit(':tell', alexaResponse);
+        /*
```
Comment (rknruben56) [inline lambda/custom/index.js]: commented out code can now be removed

File: lambda/custom/test/controllers/IntentControllerTest.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+        .reply(200, responseEvents);
+        
+    });
+
+    afterEach(function() {
+        sandbox.restore();
+    });
+
+    // Tests the searchEventsNearMe method
+    it('In: returns correct Alexa Response', async function(){
```
Comment (rknruben56) [inline lambda/custom/test/controllers/IntentControllerTest.js]: should have the function name you're testing

File: lambda/custom/test/handlers/events/EventsHandlerTest.js [hunk 6]
```diff
@@ -48,4 +48,10 @@ describe('EventsHandler Tests', function() {
             done();
         });
     });
+
```
Comment (rknruben56) [inline lambda/custom/test/handlers/events/EventsHandlerTest.js:51]: test on line 41 can now be removed

**Review by rknruben56 — CHANGES_REQUESTED** [review]
Awesome job guys. Can't believe this actually works. I would just clean it up and then it's ready to merge

**Comment by baumannalexj** [comment]
since some functions still use the non async functions, maybe we leave them until all the refactoring is done, release
then remove the unused code

**Review by baumannalexj — APPROVED** [review]

**Pull request merged**